File: api/teachPlan/src/main/java/com/sixgod/teachPlan/service/LessonService.java
package com.sixgod.teachPlan.service;

import com.sixgod.teachPlan.entity.Lesson;

public interface LessonService {
    /**
     * 根据id更新课时
     * @param lesson
     * @param id
     */
    void update(Lesson lesson, Long id);

    /**
     * 根据id获取课时
     * @param id
     * @return
     */
    Lesson findById(Long id);
}

File: web/teachPlan/src/app/service/lesson.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from "@angular/common/http";
import {Observable} from "rxjs";
import {Lesson} from "../entity/lesson";

@Injectable({
  providedIn: 'root'
})
export class LessonService {
  baseUrl = 'lesson';
  constructor(private http: HttpClient) { }

  // 更新课时
  update(lesson: Lesson, id: number):Observable<void> {
    return this.http.put<void>(this.baseUrl + '/' + id, lesson);
  }
}
